rest-api/alembic/versions/106ecd33248b_participant_views.py

```diff
@@ -0,0 +1,321 @@
+   ps.num_completed_baseline_ppi_modules,
+   ps.num_completed_ppi_modules,
+   ps.num_baseline_samples_arrived,
+   ps.samples_to_isolate_dna,
+   ps.consent_for_cabor,
+   YEAR(ps.consent_for_cabor_time) consent_for_cabor_year,
+   (SELECT IFNULL(GROUP_CONCAT(
+      IF(ac.value = 'WhatRaceEthnicity_RaceEthnicityNoneOfThese',
+         'NoneOfThese',
+         TRIM(LEADING 'WhatRaceEthnicity_' FROM
+              TRIM(LEADING 'PMI_' FROM ac.value)))),
+    'None')
+    FROM questionnaire_response qr, questionnaire_response_answer qra,
+      questionnaire_question qq, code c, code ac
+    WHERE qra.end_time IS NULL AND
+          qr.questionnaire_response_id = qra.questionnaire_response_id AND
+          qra.question_id = qq.questionnaire_question_id AND
+          qq.code_id = c.code_id AND c.value = 'Race_WhatRaceEthnicity' AND
+          qr.participant_id = p.participant_id AND
+          qra.value_code_id = ac.code_id AND
+          ac.value != 'WhatRaceEthnicity_Hispanic'
+   ) race_codes,
+   (SELECT COUNT(ac.value)
+    FROM questionnaire_response qr, questionnaire_response_answer qra,
+      questionnaire_question qq, code c, code ac
+    WHERE qra.end_time IS NULL AND
+      qr.questionnaire_response_id = qra.questionnaire_response_id AND
+      qra.question_id = qq.questionnaire_question_id AND
+      qq.code_id = c.code_id AND c.value = 'Race_WhatRaceEthnicity' AND
+      qr.participant_id = p.participant_id AND
+      qra.value_code_id = ac.code_id AND
+      ac.value = 'WhatRaceEthnicity_Hispanic'
+   ) hispanic
+ FROM
+   participant p
+     LEFT OUTER JOIN hpo ON p.hpo_id = hpo.hpo_id
+     LEFT OUTER JOIN participant_summary ps ON p.participant_id = ps.participant_id
+     LEFT OUTER JOIN code state_code ON ps.state_id = state_code.code_id
+     LEFT OUTER JOIN code recontact_method_code ON ps.recontact_method_id = recontact_method_code.code_id
+     LEFT OUTER JOIN code language_code ON ps.language_id = language_code.code_id
+     LEFT OUTER JOIN code gender_code ON ps.gender_identity_id = gender_code.code_id
+     LEFT OUTER JOIN code sex_code ON ps.sex_id = sex_code.code_id
+     LEFT OUTER JOIN code sexual_orientation_code ON ps.sexual_orientation_id = sexual_orientation_code.code_id
+     LEFT OUTER JOIN code education_code ON ps.education_id = education_code.code_id
+     LEFT OUTER JOIN code income_code ON ps.income_id = income_code.code_id
+     WHERE (ps.email IS NULL OR ps.email NOT LIKE '%@example.com') AND
+           (hpo.name IS NULL OR hpo.name != 'TEST')
+"""
+
+_PARTICIPANT_VIEW_SQL = """
+CREATE OR REPLACE VIEW participant_view AS
+ SELECT
+   p.participant_id,
+   p.sign_up_time,
+   p.withdrawal_status,
+   p.withdrawal_time,
+   p.suspension_status,
+   p.suspension_time,
+   hpo.name hpo,
+   ps.zip_code,
+   state_code.value state,
+   recontact_method_code.value recontact_method,
+   language_code.value language,
+   LEAST(89, TIMESTAMPDIFF(YEAR, ps.date_of_birth, CURDATE())) age_years,
```

Comment:
This was another transform I made on ppi_particpant_view

Reply:
(clamping age at 89); may want to revert that here as well.

Reply:
Done.